### PR TITLE
enh(conf) flag pollers as updated when changing host/hostgroup/service/servicegroup configuration

### PR DESCRIPTION
**Pull request by a-launois**
## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/centreon-clapi/centreonConfiguration.class.php [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+            . implode(', ', array_keys($bindedParams)) . ")";
+        $stmt = $this->db->prepare($query);
+        foreach ($bindedParams as $bindedParam => $bindedValue) {
+            $stmt->bindValue($bindedParam, $bindedValue, \PDO::PARAM_INT);
+        }
+        $stmt->execute();
+    }
+
+    /**
+     * Undocumented function
```
Comment (jeremyjaouen) [inline www/class/centreon-clapi/centreonConfiguration.class.php]:
```suggestion
     * Undocumented function
```
document this function

File: www/class/centreon-clapi/centreonHost.class.php [hunk 2]
```diff
@@ -374,6 +393,61 @@ public function del($objectName)
             "DELETE FROM service WHERE service_register = '1' "
             . "AND service_id NOT IN (SELECT service_service_id FROM host_service_relation)"
         );
+
+        $centreonConfig->signalConfigurationChange(self::RESOURCE_TYPE, $hostId, $previousPollerIds);
+    }
+
+    /**
+     * @param array $parameters
```
Comment (jeremyjaouen) [inline www/class/centreon-clapi/centreonHost.class.php]: precise type of data

File: www/class/centreon-clapi/centreonService.class.php [hunk 3]
```diff
@@ -316,6 +319,20 @@ public function show($parameters = null, $filters = array())
         }
     }
 
+
```
Comment (jeremyjaouen) [inline www/class/centreon-clapi/centreonService.class.php]:
```suggestion
```

File: www/class/centreon-clapi/centreonServiceGroup.class.php [hunk 4]
```diff
@@ -407,9 +431,70 @@ public function __call($name, $arg)
...
+
         $parentDependency = new \Centreon_Object_DependencyServicegroupParent($this->dependencyInjector);
         $parentDependency->removeRelationLastServicegroupDependency($sgId);
         parent::del($objectName);
+
+        $centreonConfig->signalConfigurationChange(self::RESOURCE_TYPE, $sgId, $previousPollerIds);
+    }
+
+    /**
+     * @param array $parameters
```
Comment (jeremyjaouen) [inline www/class/centreon-clapi/centreonServiceGroup.class.php]:
```suggestion
add type of data
```

File: www/include/configuration/configObject/host/DB-Func.php [hunk 5]
```diff
@@ -268,13 +268,20 @@ function enableHostInDB($host_id = null, $host_arr = array())
...
-        $dbResult = $pearDB->query("UPDATE host SET host_activate = '1' WHERE host_id = '" . (int)$key . "'");
-        $dbResult2 = $pearDB->query("SELECT host_name FROM `host` WHERE host_id = '" . (int)$key . "' LIMIT 1");
-        $row = $dbResult2->fetch();
-        $centreon->CentreonLogAction->insertLog("host", $key, $row['host_name'], "enable");
+    foreach (array_keys($host_arr) as $hostId) {
+        $stmt = $pearDB->prepare("UPDATE host SET host_activate = '1' WHERE host_id = :hostId");
+        $stmt->bindValue(':hostId', $hostId, \PDO::PARAM_INT);
+        $stmt->execute();
+
+        $stmt = $pearDB->prepare("SELECT host_name FROM `host` WHERE host_id = :hostId");
```
Comment (a-launois) [inline www/include/configuration/configObject/host/DB-Func.php]: place prepare outside foreach, idem for others

File: www/class/centreon-clapi/centreonConfigurationChange.class.php [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+    public const RESOURCE_TYPE_SERVICEGROUP = 'servicegroup';
+
+    /**
+     * Constructor
+     *
+     * @return void
+     */
+    public function __construct(
+        private \CentreonDB $db
+    ) {}
```
Comment (callapa) [inline www/class/centreon-clapi/centreonConfigurationChange.class.php]:
```suggestion
    ) {
    }
```

**Review by callapa — APPROVED** [review]

File: www/include/common/common-Func.php [hunk 7]
```diff
@@ -2498,3 +2498,291 @@ function unvalidFormMessage()
...
+                getHostsForConfigChangeFlagFromServiceIds([$resourceId], $shouldResourceBeEnabled)
+            );
+            break;
+        case 'servicegroup':
+            $hostIds = array_merge(
+                $hostIds,
+                getHostsForConfigChangeFlagFromServiceGroupId($resourceId, $shouldResourceBeEnabled)
+            );
+            break;
+    }
```
Comment (callapa) [inline www/include/common/common-Func.php:2784]: No default option ?

**Review by jeremyjaouen — DISMISSED** [review]

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**